### PR TITLE
Refactor e2e tests

### PR DESCRIPTION
**Pull request by cimnine**
## Summary

This change aims to simplify the actual test cases and allows
to extend them more granuarly, e.g. by allowing to
re-use the resource definitions by composing them as needed.

The idea is that all the e2e tests can use the resource definitions in the new `definitions` folder to their liking.
Otherwise, e.g. the `deployment.yaml`, would have to be duplicated for every test.
Or the upcoming test of a restore will probably have to do a `backup` as well. So it can share that with the e2e test that tests a backup.

Related to #272.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`,
      as they show up in the changelog
- [x] ~Update the documentation.~
- [x] ~Update tests.~
- [x] ~Link this PR to related issues.~

### CODE REVIEW

File: e2e/lib/k8up.bash [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+	cp -r /tmp/detik debug || true
+}
+
+kustomize() {
+	go run sigs.k8s.io/kustomize/kustomize/v3 "${@}"
+}
+
+prepare() {
+	mkdir -p "debug/${1}"
+	kustomize build "${1}" > "debug/${1}/main.yml"
```
Comment (ccremer) [inline e2e/lib/k8up.bash:20]:
```suggestion
	kustomize build "${1}" -o "debug/${1}/main.yml"
```

File: e2e/test-deployment.bats [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+load "lib/detik"
+load "lib/k8up"
+
+DETIK_CLIENT_NAME="kubectl"
+DETIK_CLIENT_NAMESPACE="k8up-system"
+DEBUG_DETIK="true"
+
+@test "reset the debug file" {
+	reset_debug
+}
```
Comment (ccremer) [inline e2e/test-deployment.bats:13]: Have you tried moving this reset to setup() or teardown() ? Seems kinda out of place as a "test case"

Comment (cimnine) [reply e2e/test-deployment.bats]: I guess this was originally copied from the DETIK example. I've moved it to a `setup_file` function in the `k8up.bash` file.

**Comment by cimnine** [comment]
Integrated feedback and rebased onto latest `master`.

**Review by ccremer — APPROVED** [review]

**Pull request merged**